server/authentication/auth.js

```diff
@@ -1,16 +1,18 @@
 const passport = require('passport')
 const { Strategy } = require('passport-oauth2')
+const querystring = require('querystring')
 const config = require('../config')
 const { generateOauthClientToken } = require('./clientCredentials')
 
-function authenticationMiddleware() {
-  return (req, res, next) => {
-    if (req.isAuthenticated()) {
+function authenticationMiddlewareFactory(tokenVerifier) {
+  return async (req, res, next) => {
+    if (req.isAuthenticated() && (await tokenVerifier.verify(res.locals.user.token))) {
       return next()
     }
-
-    req.session.returnTo = req.originalUrl
```

Comment:
this is used by `/login/callback` in app.js. Do we need to remove/change that?

Reply:
No